.github/workflows/update-and-rebuild.yaml

```diff
@@ -1,60 +1,124 @@
-        required: true
-        type: string
-
-  workflow_call:
-    inputs:
-      osmosis_rev:
-        required: true
-        type: string
-
-env:
-  CARGO_TERM_COLORS: always
+  schedule:
+    # Every day
+    - cron: "0 1 * * *"
 
 jobs:
-  build:
+  check-new-finschia-tag:
     runs-on: ubuntu-latest
+    outputs:
+      NEW_TAG: ${{ env.NEW_TAG }}
+      FINSCHIA_SDK_VERSION: ${{ steps.get_versions.outputs.FINSCHIA_SDK_VERSION }}
+      WASMD_VERSION: ${{ steps.get_versions.outputs.WASMD_VERSION }}
+      IBC_GO_VERSION: ${{ steps.get_versions.outputs.IBC_GO_VERSION }}
+      TENDERMINT_VERSION: ${{ steps.get_versions.outputs.TENDERMINT_VERSION }}
     steps:
-      - name: Setup Go
-        uses: actions/setup-go@v3
+      - name: Set current tag to env
+        run: |
+          CURRENT_TAG=$(curl  "https://api.github.com/repos/Finschia/finschia-proto/tags" | jq -r '.[0].name')
+          echo "CURRENT_TAG=$CURRENT_TAG" >> $GITHUB_ENV
+
+      - name: Check out to finschia
+        uses: actions/checkout@v4
         with:
-          go-version: ">=1.19.0"
+          repository: Finschia/finschia
+          fetch-depth: 0
 
-      - name: Install buf
+      - name: Check new tag released
         run: |
-          GO111MODULE=on go install github.com/bufbuild/buf/cmd/buf@v1.15.1
+          NEW_TAG=$(curl "https://api.github.com/repos/Finschia/finschia/tags" | jq -r '.[0].name')
+          if [ ${{ env.CURRENT_TAG }} != $NEW_TAG ] && \
+          [ "$(git branch -r | grep "auto-bump-up/${NEW_TAG}")" == "" ]; then \
+            echo "NEW_TAG=$NEW_TAG" >> $GITHUB_ENV; \
+            git checkout $NEW_TAG
+          fi
+
+      - name: get each submodule version from go.mod in finschia
+        if: env.NEW_TAG
+        id: get_versions
+        run: |
+          FINSCHIA_SDK_VERSION=$(grep 'github.com/Finschia/finschia-sdk' go.mod | awk '{print $2}' | awk -F '-' '{if ($3 != "") {print $3} else  if ($2 != "") {print $1"-"$2} else {print $1}}')
+          WASMD_VERSION=$(grep 'github.com/Finschia/wasmd' go.mod | awk '{print $2}' | awk -F '-' '{if ($3 != "") {print $3} else  if ($2 != "") {print $1"-"$2} else {print $1}}')
+          IBC_GO_VERSION=$(grep 'github.com/cosmos/ibc-go' go.mod | grep -v '=>' | awk '{print $2}' | awk -F '-' '{if ($3 != "") {print $3} else  if ($2 != "") {print $1"-"$2} else {print $1}}')
+          TENDERMINT_VERSION=$(grep 'github.com/tendermint/tendermint' go.mod | grep -v '=>' | awk '{print $2}' | awk -F '-' '{if ($3 != "") {print $3} else  if ($2 != "") {print $1"-"$2} else {print $1}}')
+          echo "NEW_TAG=${NEW_TAG}" >> $GITHUB_OUTPUT
+          echo "FINSCHIA_SDK_VERSION=$FINSCHIA_SDK_VERSION" >> $GITHUB_OUTPUT
+          echo "WASMD_VERSION=$WASMD_VERSION" >> $GITHUB_OUTPUT
+          echo "TENDERMINT_VERSION=$TENDERMINT_VERSION" >> $GITHUB_OUTPUT
+          echo "IBC_GO_VERSION=$IBC_GO_VERSION" >> $GITHUB_OUTPUT
+
+  create-bump-up-PR:
+    runs-on: ubuntu-latest
+    needs: check-new-finschia-tag
```

Comment:
This job only works when `env.NEW_TAG` is not null, so you can add an `if` condition to the job, and this can reduce the number of if conditions in each step.

Reply:
This action is also used to modify the "packages/proto-build" code, so it is not limited to NEW_TAG.

Reply:
Sorry. I missunderstood. OK. I fixed here.
[fix: Change to perform "if: needs.check-new-finschia-tag.outputs.NEW_…](https://github.com/Finschia/finschia-wasm/pull/9/commits/b096dbcef769619315c784b00cf47e5136bc7c38)

Reply:
fixed.
[fix: swap the order of "job dependency definition" and "job dependenc…](https://github.com/Finschia/finschia-wasm/pull/9/commits/927e88ca2ab0d9894b99ee44e4f0c5fb496c85ae)